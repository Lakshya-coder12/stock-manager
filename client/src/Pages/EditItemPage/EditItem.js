import React, { Component } from "react";
import Button from "../../Components/Button/Button";
// import { connect } from "react-redux";
// import {
//   fetchSingleItem,
//   removeSingleItem,
//   updateItem,
// } from "../../redux/actions/ItemActions";
import PropTypes from "prop-types";
import jwt_decode from "jwt-decode";

class EditItem extends Component {
  constructor(props) {
    super(props);
    this.itemID = this.props.match.params;
    this.state = {
      name: "",
      companyName: "",
      stock: 0,
      username: "",
    };
  }
  handleInputChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };
  handleSubmit = (e) => {
    e.preventDefault();
    const data = {
      name: this.state.name,
      companyName: this.state.companyName,
      stock: this.state.stock,
    };
    this.props.updateItem(this.itemID.id, data, this.props.history);
  };
  handleRedirect = () => {
    this.props.history.push("/home");
  };

  handleLogout = () => {
    this.props.logout(this.props.history);
  };

  componentDidMount() {
    const token = localStorage.getItem("jwt");
    if (!token) {
      this.props.history.push("/login");
    } else {
      const user = jwt_decode(token);
      const username = user.user.name;
      this.setState({
        username: username,
      });
      this.props.fetchSingleItem(this.itemID.id);
      this.setState(this.props.singleItem.item);
    }
  }
  componentWillUnmount() {
    this.props.removeSingleItem();
  }
  render() {
    // const user = this.props.user.user;
    return (
      <>
        <Button className="abs-button" onClick={this.handleRedirect}>
          Home
        </Button>
        <form className="form" onSubmit={this.handleSubmit}>
          <div className="input-element">
            <label className="label">Name</label>
            <input
              required
              name="name"
              onChange={this.handleInputChange}
              value={this.state.name}
              className="input"
              type="text"
            />
          </div>
          <div className="input-element">
            <label className="label">Company Name</label>
            <input
              required
              name="companyName"
              onChange={this.handleInputChange}
              value={this.state.companyName}
              className="input"
              type="text"
            />
          </div>
          <div className="input-element">
            <label className="label">Stock</label>
            <input
              required
              name="stock"
              onChange={this.handleInputChange}
              value={this.state.stock}
              className="input"
              type="number"
              step={1}
            />
          </div>
          <div className="form-btn">
            <Button>Update Item</Button>
          </div>
        </form>
        <div className="logout">
          <span className="username">{this.state.username}</span>
          <Button onClick={this.handleLogout}>Logout</Button>
        </div>
      </>
    );
  }
}
EditItem.propTypes = {
  fetchSingleItem: PropTypes.func,
  updateItem: PropTypes.func,
  removeSingleItem: PropTypes.func,
  logout: PropTypes.func,
};
// const mapStateToProps = (state) => {
//   return {
//     singleItem: state.singleItem,
//   };
// };

// const mapDispatchToProps = (dispatch) => {
//   return {
//     fetchSingleItem: (id) => dispatch(fetchSingleItem(id)),
//     updateItem: (id, data, history) => dispatch(updateItem(id, data, history)),
//     removeSingleItem: () => dispatch(removeSingleItem()),
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(EditItem);
export default EditItem;
