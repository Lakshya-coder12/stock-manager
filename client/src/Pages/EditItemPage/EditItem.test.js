import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import EditItem from "./EditItem";

configure({ adapter: new Adapter() });

describe("Testing Edit Item page", () => {
  it("matches snapshot", () => {
    const paramsMock = jest.fn();
    const itemMock = jest.fn();
    const fetchSingleItemMock = jest.fn();
    const pushMock = jest.fn();
    const updateItemMock = jest.fn();
    const removeSingleItemMock = jest.fn();
    const wrapper = shallow(
      <EditItem
        singleItem={itemMock}
        fetchSingleItem={fetchSingleItemMock}
        updateItem={updateItemMock}
        history={{ push: pushMock }}
        removeSingleItem={removeSingleItemMock}
        match={{ params: paramsMock }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
