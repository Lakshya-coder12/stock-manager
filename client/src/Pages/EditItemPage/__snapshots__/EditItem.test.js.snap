// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Edit Item page matches snapshot 1`] = `
<Fragment>
  <Button
    className="abs-button"
    onClick={[Function]}
  >
    Home
  </Button>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Name
      </label>
      <input
        className="input"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Company Name
      </label>
      <input
        className="input"
        name="companyName"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Stock
      </label>
      <input
        className="input"
        name="stock"
        onChange={[Function]}
        required={true}
        step={1}
        type="number"
        value={0}
      />
    </div>
    <div
      className="form-btn"
    >
      <Button>
        Update Item
      </Button>
    </div>
  </form>
  <div
    className="logout"
  >
    <span
      className="username"
    />
    <Button
      onClick={[Function]}
    >
      Logout
    </Button>
  </div>
</Fragment>
`;
